src/TestUtils/src/DeviceTests/AssertionExtensions.Windows.cs

```diff
@@ -49,42 +50,50 @@ public static Task SendKeyboardReturnType(this FrameworkElement view, ReturnType
 		public static async Task WaitForFocused(this FrameworkElement view, int timeout = 1000)
 		{
 			TaskCompletionSource focusSource = new TaskCompletionSource();
-			view.GotFocus += OnFocused;
+
+			FocusManager.GotFocus += OnFocused;
 
 			try
 			{
 				await focusSource.Task.WaitAsync(TimeSpan.FromMilliseconds(timeout));
 			}
 			finally
 			{
-				view.GotFocus -= OnFocused;
+				FocusManager.GotFocus -= OnFocused;
 			}
 
-			void OnFocused(object? sender, RoutedEventArgs e)
+			void OnFocused(object? sender, FocusManagerGotFocusEventArgs e)
 			{
-				view.GotFocus -= OnFocused;
-				focusSource.SetResult();
+				if (e.NewFocusedElement == view)
+				{
+					FocusManager.GotFocus -= OnFocused;
+					focusSource.SetResult();
```

Comment:
Do we care about setting the result more than once? If don't I suggest changing it to be `TrySetResult`. `SetResult` will throw an exception if it's already set

Reply:
I don't think that could happen considering the event is being unsubscribed in the line above.

If there was a race condition, raising an exception would be exactly what we want in this device test.

Reply:
My point is, if that happen, do we care about it? If an exception throws here I think it will break the app, since the developer can't, easily, add a try/catch here. Is this exception important enough to break an app or is OK to just ignore it?

Reply:
Ooh my bad, I didn't see it's a unit test. Reviewing code from mobile is hard 😅

Reply:
@MartyIX yes, and it looks like to have a good gotcha on here, so I'm interested in learning new tricks (;